File: Basics/0.6-Nested-Loops/Lab/cinema_tickets.py
total_tickets = 0
total_student_tickets = 0
total_standard_tickets = 0
total_kid_tickets = 0

movie = input()
while movie != "Finish":
    free_seats = int(input())
    total_free_seats = free_seats
    current_total_tickets = 0

    while free_seats > 0:
        ticket_type = input()

        if ticket_type == "End":
            break
        elif ticket_type == "student":
            total_student_tickets += 1
        elif ticket_type == "standard":
            total_standard_tickets += 1
        elif ticket_type == "kid":
            total_kid_tickets += 1

        current_total_tickets += 1
        total_tickets += 1
        free_seats -= 1

    current_room_volume = current_total_tickets / total_free_seats * 100

    print(f"{movie} - {current_room_volume:.2f}% full.")

    movie = input()

print(f"Total tickets: {total_tickets}")
print(f"{total_student_tickets / total_tickets * 100:.2f}% student tickets.")
print(f"{total_standard_tickets / total_tickets * 100:.2f}% standard tickets.")
print(f"{total_kid_tickets / total_tickets * 100:.2f}% kids tickets.")

File: JS-Basics/0-6-Nested-Loops/Lab/cinema-tickets.js
function cinemaTickets(input) {
    let standardTickets, kidTickets, studentTickets, occupiedChairs;
    [standardTickets, kidTickets, studentTickets, occupiedChairs] = [0, 0, 0, 0];
    while (input[0] != "Finish") {
        occupiedChairs = 0;
        let movie = input.shift(); 
        let freeChairs = Number(input.shift());
        while (occupiedChairs < freeChairs && input[0] != "End" && input[0] != "Finish") {
            let ticket = input.shift(); 
            if (ticket == "standard") {standardTickets++;}
            else if (ticket == "kid") {kidTickets++;}
            else if (ticket == "student") {studentTickets++;}
            occupiedChairs++;
        }
        occupiedChairs = occupiedChairs / freeChairs * 100;
        console.log(`${movie} - ${occupiedChairs.toFixed(2)}% full.`);
        if (input[0] == "End") {input.shift()}
    }
    
    totalTickets = standardTickets + studentTickets + kidTickets;
    standardTickets = standardTickets / totalTickets * 100;
    studentTickets = studentTickets / totalTickets * 100;
    kidTickets = kidTickets / totalTickets * 100;
    console.log(`Total tickets: ${totalTickets}`);
    console.log(`${studentTickets.toFixed(2)}% student tickets.`);
    console.log(`${standardTickets.toFixed(2)}% standard tickets.`);
    console.log(`${kidTickets.toFixed(2)}% kids tickets.`);
}


cinemaTickets(["Taxi",
"10",
"standard",
"kid",
"student",
"student",
"standard",
"standard",
"End",
"Scary Movie",
"6",
"student",
"student",
"student",
"student",
"student",
"student",
"Finish"]);
cinemaTickets(["The Matrix",
"20",
"student",
"standard",
"kid",
"kid",
"student",
"student",
"standard",
"student",
"End",
"The Green Mile",
"17",
"student",
"standard",
"standard",
"student",
"standard",
"student",
"End",
"Amadeus",
"3",
"standard",
"standard",
"standard",
"Finish"]);